docs/logs/getting-started/Program.cs

```diff
@@ -15,17 +15,40 @@
 // </copyright>
 
 using System.Collections.Generic;
+#if NETCOREAPP2_1
+using Microsoft.Extensions.DependencyInjection;
+#endif
 using Microsoft.Extensions.Logging;
+using OpenTelemetry;
 
 public class Program
 {
     public static void Main()
     {
+        /*
+        The OpenTelemetry style:
```

Comment:
Though this style puts logging more aligned with Otel Tracing setup, this distances us from general ILogger guidelines.
Some thoughts:
If we go with this route, then users will have to rewire their entire logging plumbing, if they ever chose to move away from OpenTelemetry. If we stick to the existing ILogger style, then adding/removing OpenTelemetry is adding/removing one line - builder.AddOtel..(). The way user obtain ilogger instances (from logger factory or DI) does not have to change at all.

Reply:
@alanwest @CodeBlanch please share your thoughts on the approaches, when you get a chance!

Reply:
@cijothomas Good callout! We definitely want to fit into the standard .net core/extensions patterns. I think the code is actually fine. The only weird thing is the build-up pattern in this example project. I took a stab at fitting it into a more standard approach:

https://github.com/open-telemetry/opentelemetry-dotnet/compare/reyang/ilogger...CodeBlanch:reyang/ilogger

Reply:
@CodeBlanch Thanks. The code from you branch is fitting the pattern where there is a host and/or DI . We want to provide pure console app example. Its not possible in .netcore2.1 as only way to setup ilogger requires DI, but for other versions, its possible without Hosting/DI.

Reply:
I suppose then I don't understand your issue with the design 😄 LoggerProvider is registered with the Factory. The Factory creates ILogger using all registered providers. What do you see that is deviating from the ILogger guidelines?

Regarding the pattern in the sample, makes sense. BUT I think it might be more confusing than useful to people. The [official guide](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-5.0) uses the host. So our example should too IMO. If we want to provide another example, that shows how to do it without the host, OK with that, but it feels like the exception, not the rule.

Reply:
This "getting-started" example is far more complex than what a real getting started example should be (I think it should be less than 10 lines of code).
My suggestion is to get it merged as is so we can move forward on the exporter design.
I can work with the owner of ILogger to see what is the recommended way of getting the ILogger instance, and send another PR to fix the example.
I think the outcome should be - whatever we suggest people here is aligned with the official document (which could also mean that we change the official document).

Reply:
Sorry i wasn't super clear earlier!
The following style of obtaining logger is deviating from Logging styles, but is well aligned with Otel style we have for traces.
```
using var loggerProvider = Sdk.CreateLoggerProviderBuilder()
            .AddProcessor(new MyProcessor())
            .Build();
        var logger = loggerProvider.CreateLogger("MyLogger"); // this line is not how Ilogger instances are typically obtained
```

In Ilogger world, the way to obtain ilogger instance is either of the options:
1. LoggerFactory.CreateLogger()
2. serviceProvider.GetRequiredService<ILogger>

Reply:
@CodeBlanch There is non-host example as well now: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-5.0#logging-providers-1

Reply:
@reyang Agree. we can iterate on this and reach the right balance.

Reply:
@cijothomas Gotcha. Agree with you we want to promote creating a logger through the LoggerFactory. Maybe we shouldn't even provide `Sdk.CreateLoggerProviderBuilder` and only have the `ILoggingBuilder` pattern?

@reyang Sounds good to me.

Reply:
Sorry, coming to this conversation late... I agree that `loggerProvider.CreateLogger` is not ideal and it's best to conform to using the `LoggerFactory.CreateLogger()` in the context of this console application.

I think I agree with @CodeBlanch that `Sdk.CreateLoggerProviderBuilder` may not be required. If I'm following everyone's comments I think we'd want an extension method off of ILoggingBuilder like:

```csharp
        public static ILoggingBuilder AddOpenTelemetry(this ILoggingBuilder builder, Action<OpenTelemetryLoggerOptions> configure)
        {
                   ...
        }
```

Posting in case you haven't read [this](https://blog.stephencleary.com/2018/06/microsoft-extensions-logging-part-2-types.html). I find Stephen Cleary's explanation of `ILogger` and friends more consumable than some of the Microsoft documentation 😄. From his blog:

> ILoggerFactory is a collection of ILoggerProviders that creates composite ILogger/ILogger<T> loggers.
ILoggerProvider ia a provider for a specific logging system. It provides ILogger loggers to the ILoggerFactory.


That is, it is expected that someone configure all of their logging related concerns through the `ILoggingBuilder` and then use `ILoggerFactory` to get `ILogger` instances. It would be very unusual for someone to use an `ILoggerProvider` directly.

Reply:
I've removed `Sdk.CreateLoggerProviderBuilder` based on the feedback here.